WalletWasabi.Backend/Global.cs

```diff
@@ -252,6 +255,12 @@ protected virtual void Dispose(bool disposing)
 
 					await P2pNode.DisposeAsync().ConfigureAwait(false);
 					Logger.LogInfo($"{nameof(P2pNode)} is disposed.");
+
+					if (WhiteList is { } whiteList)
+					{
+						await whiteList.WriteToFileIfChangedAsync().ConfigureAwait(false);
+						Logger.LogInfo($"{nameof(WhiteList)} is saved to file.");
+					}
```

Comment:
I don't understand this.

In the beginning the `Whitelist Guid`s will be the same, so we will not save to file.

And `WriteToFileIfChangedAsync` is not guaranteed to write to file, so IMO the log message here is misleading.

Consider moving this log line inside `WriteToFileIfChangedAsync`.

Reply:
Correct.